### PR TITLE
add CRUD for creating a new business

### LINKED ISSUE #5: A business can be created by a registered user
Visit the page to create a new business 
Enter the business's name and description
click button "Create Business"
expect the page to be the home page for the business 
with a flash message that the business has been created

### PR DESCRIPTION
**Pull request by saylerb**
Okay so I added the ability to create a new business. Some items to consider:

We need to handle how the app handles new _but pending_ businesses. Right now, you can create an inactive business and still navigate to that business' show page. Not sure if we want that behavior. I'll create an issue for this so we don't forget.

closes #5 

connects to #120

### CODE REVIEW

File: app/controllers/businesses_controller.rb [hunk 1]
```diff
@@ -26,8 +26,30 @@ def update
...
   end
 
+  def new
+    @business = Business.new
+  end
+
+  def create
+    user = current_user
+    @business = Business.new(new_business_params)
+    user.businesses << @business
```
Comment (ErinGreenhalgh) [inline app/controllers/businesses_controller.rb:36]: change to be more secure:
@business = current_user.businesses.create(new_business_params)

File: spec/integration/user_can_create_a_new_business_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+             with: "A vertically integrated subscription snack service"
+
+    click_button "Create Business"
+
+    new_business = Business.last
+
+    expect(new_business.admins.include?(user)).to be true
+
+    visit business_path(new_business.slug)
+  end
```
Comment (ErinGreenhalgh) [inline spec/integration/user_can_create_a_new_business_spec.rb:35]: let's add an assertion that they see a pending message on the business dashboard when it has been created but not yet platform-admin approved

**Comment by saylerb** [comment]
I'm going to make a couple changes here: 
1. make creation of a business more secure in controller.
2. Add the display of a business show + the business admin dashboard to be "Pending"

File: app/controllers/businesses_controller.rb [hunk 3]
```diff
@@ -26,8 +26,30 @@ def update
...
+
+    if @business.save
+      flash[:success] = "Business successfully created!"
+      redirect_to dashboard_path
+    else
+      flash.now[:error] = @business.errors.full_messages
```
Comment (ErinGreenhalgh) [inline app/controllers/businesses_controller.rb:42]: we'll probably want to join these on a comma so they are more readable

File: app/views/business/dashboard/show.html.erb [hunk 4]
```diff
@@ -3,11 +3,15 @@
...
 
+    <% if @business.items.exists? %>
+      <%= render partial: 'open_auctions_table' %>
+      <%= render partial: 'closed_auctions_table' %>
+    <% else  %>
+      <p>There are no items for this business yet.</p>
+    <% end  %>
+  <% else %>
+    <p id="pending_notice">This business is pending approval.</p>
+  <% end %>
```
Comment (ErinGreenhalgh) [inline app/views/business/dashboard/show.html.erb:16]: let's add this to our list to think about ways to refactor out the nested if

File: config/routes.rb [hunk 5]
```diff
@@ -6,15 +6,13 @@
...
   #resources :items, only: [:index, :show]
   resources :users, only: [:new, :index, :create]
   get "/user/edit", to: "users#edit"
   patch "/user/edit", to: "users#update"
 
-
-  resources :businesses, only: [:index]
+  resources :businesses, only: [:index, :new, :create]
 
   # get '/admin-dashboard', to: 'business_admins#show', as: 'admin_dashboard'
```
Comment (ErinGreenhalgh) [inline config/routes.rb:17]: we can delete this line!

**Pull request merged**